integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateTableHelper.scala

```diff
@@ -110,7 +110,29 @@ case class PreAggregateTableHelper(
     // Datamap table name and columns are automatically added prefix with parent table name
     // in carbon. For convenient, users can type column names same as the ones in select statement
     // when config dmproperties, and here we update column names with prefix.
-    val longStringColumn = tableProperties.get(CarbonCommonConstants.LONG_STRING_COLUMNS)
```

Comment:
emm, actually this line of code was added to solve the same problem mentioned in the PR description. Previously we thought it's user's responsibility to explicitly specify the long_string_columns in the preagg datamap.

Please @kevinjmh also check this.

Reply:
This PR is for the scenario when user doesn't config the long_string_columns in dmprop then we take long_string_cols from the parent table.